cli/pcluster/cw-dashboard-substack.cfn.yaml

```diff
@@ -0,0 +1,253 @@
+                    - '{{ i|int }}'
+                    - !Ref 'EBSVolumesIDs'
+                    {%- endif %}
+                  {%- endfor %}
+                  {{ update_position(6,6) }}
+              - ']}}'
+                {% endif %} {# if isPresent.bool #}
+              {%- endfor %}
+              {{ update_pos_after_section(6) }}
+
+              {#- Unconditional RAID metrics #}
+              # RAID metrics graphs
+              {%- set number_of_raid_volumes = config.RAIDOptions.split(',')[2]|int -%} {# if NONE, converts to 0! #}
+              {%- if number_of_raid_volumes > 0 %}
+              - !Join # TODO: remove this join that is not needed anymore since we do not use the CreateRAIDSubstack condition
+                - ''
+                - - ',{"type":"text","x":{{ coord.x }},"y":{{ coord.y }},"width":24,"height":1,"properties":{"markdown":"\n# RAID Metrics\n"}}'
+                  {{ update_pos_after_section(1) }}
+                  {%- set raid_metrics = [["VolumeReadOps", "VolumeWriteOps"], ["VolumeReadBytes", "VolumeWriteBytes"],
+                                         ["VolumeTotalReadTime", "VolumeTotalWriteTime"], ["VolumeQueueLength"],
+                                         ["VolumeIdleTime"]] %}
+                  {%- set raid_volume_type = config.RAIDOptions.split(',')[3] %}
+
+                  {%- for metrics_group in raid_metrics %}
+
+                  # RAID {{ metrics_group|join(', ') }} graph
+                  - !Sub ',{"type":"metric","x":{{ coord.x }},"y":{{ coord.y }},"width":6,"height":6,"properties":{"view":"timeSeries","stacked":false,"region":"${AWS::Region}","metrics":'
+                    {%- for i in range(number_of_raid_volumes) %}
+                      {%- set volume_loop = loop %}
+                      {%- for metric in metrics_group %}
+                  - !Sub
+                    - '{% if volume_loop.first and loop.first %}[{% else %},{% endif %}["AWS/EBS","{{ metric }}","VolumeId","${RAID_Volume{{ i|int +1 }}}"]{% if volume_loop.last and loop.last %}]}}{% endif %}'
+                    - RAID_Volume{{ i|int +1 }}: !Select
+                        - '{{ i|int }}'
+                        - !Ref 'RAIDVolumesIDs'
+                      {%- endfor %}
+                    {%- endfor %}
+                    {{ update_position(6,6) }}
+                  {%- endfor %}
+
+                  {# Conditional RAID metrics -#}
+                  {%- set raid_metrics_conditions = [{'metric': 'VolumeConsumedReadWriteOps', 'accepted_vol_types': ["io1"]},
+                                                    {'metric': 'VolumeThroughputPercentage', 'accepted_vol_types': ["io1"]},
+                                                    {'metric': 'BurstBalance', 'accepted_vol_types': ["gp2", "st1", "sc1"]}]
+                  %}
+                  {%- for metric_condition in raid_metrics_conditions %}
+                    {%- if raid_volume_type in metric_condition.accepted_vol_types %}
```

Comment:
It seems to me we can unify the conditional and unconditionals loops, by adding something like:
```python
if raid_volume_type in metric_condition.accepted_vol_types || not metric_condition.accepted_vol_types
```

but let's eventually keep this improvement as latest step

Reply:
[STILL TODO] I do not know if it would be very clear, we should discuss later about this, keeping as improvement at the moment